src/types/advertisement.rs

```diff
@@ -79,6 +85,12 @@ impl Advertisement<'_> {
             CompleteLocalName(n) => {
                 bytes[2..2 + n.len()].copy_from_slice(n.as_bytes());
             }
+            CompleteListOf16BitServices(n) => {
+                LittleEndian::write_u16_into(n, &mut bytes[2..2+(n.len() * 2)]);
+            }
+            CompleteListOf128BitServices(n) => {
+                LittleEndian::write_u128_into(n, &mut bytes[2..2+(n.len() * 16)]);
+            }
```

Comment:
I think using `core::mem::sizeof::<u16>` and `core::mem::sizeof::<u128>` might be better than using magic numbers.

Reply:
I agree, but this is remaining consistent with the rest of the codebase. It would be a little strange to use that here but then have magic numbers everywhere else.